doc/sphinx_source/mainDocs/pbkdf2.rst

```diff
@@ -0,0 +1,94 @@
+Last revised: October 28, 2020
+
+**************
+PBKDF2 Hashing
+**************
+
+    With the release of Eggdrop 1.9.0, the development team is releasing an updated crytopgraphy module, PBKDF2, for use with Eggdrop. PBKDF2 is a one-way hashing algorithm used to protect the contents of the password file, as well as for use via the Tcl interface.
+
+Background
+==========
+    Prior to Eggdrop 1.9.0, the blowfish module was included with Eggdrop to protect sensitve data such as passwords stored in the userfile. While there are no known practical attacks against blowfish at the time of this writing, it was decided that a more modern crypto solution was desirable to be included with Eggdrop. The PBKDF2 (Password-based Key Derivation Function 2) uses a password and salt value to create a password hash (the salt value ensures that the hashes of two identical passwords are different). This process is one-way, which means the hashes cannot be cryptographically reversed and thus are safe for storing in a file.
+
+    The default configuration of Eggdrop 1.9.0 has both the blowfish and pbkdf2 modules enabled (see `Hybrid Configuration`_ below). This will allow users upgrading a seamless transition to the PBKDF2 module. For users starting an Eggdrop for the first time, it is recommended to comment out the 'loadmodule blowfish' line, in order to implement the `Solo Configuration`_.
+
+    Also of note, the maximum password length is increased to 30 with PBKDF2, up from 15 with Blowfish. The automatically-generated botnet passwords are now 30 characters instead of the maximum-allowed 16 used with blowfish the blowfish module, and pull from a larger character set than what was used with the blowfish module. Finally, if you are linking bots, you'll need to ensure the same module is loaded on both bots (ie, if the hub bot is using the pbkdf2 module, the leaf bots must have pbkdf2 loaded as well in order to enable authentication checks).
```

Comment:
Word repeat typo: "blowfish the blowfish". Also "pull" should be past tense, shouldnt it?